src/main/java/seedu/address/model/eatery/EateryAttributesContainsKeywordsPredicate.java

```diff
@@ -0,0 +1,59 @@
+package seedu.address.model.eatery;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import seedu.address.commons.util.StringUtil;
+
+/**
+ * Tests that a {@code Eatery}'s attributes ({@code Name}, {@code Address}, {@code Category}, {@code Tags})
+ * matches any of the keywords given.
+ */
+public class EateryAttributesContainsKeywordsPredicate implements Predicate<Eatery> {
+    private final List<String> nameKeywords;
+    private final List<String> addressKeywords;
+    private final List<String> categoryKeywords;
+    private final List<String> tagKeywords;
+
+    public EateryAttributesContainsKeywordsPredicate(List<String> nameKeywords, List<String> addressKeywords,
+                                                     List<String> categoryKeywords, List<String> tagKeywords) {
+        this.nameKeywords = nameKeywords;
+        this.addressKeywords = addressKeywords;
+        this.categoryKeywords = categoryKeywords;
+        this.tagKeywords = tagKeywords;
+    }
+
+    public EateryAttributesContainsKeywordsPredicate(List<String> nameKeywords) {
+        this.nameKeywords = nameKeywords;
+        this.addressKeywords = new ArrayList<>();
+        this.categoryKeywords = new ArrayList<>();
+        this.tagKeywords = new ArrayList<>();
+    }
```

Comment:
What will happen in the event that the user input any other attributes instead of name?